controllers/wavefrontalert_controller.go

```diff
@@ -105,39 +133,123 @@ func (r *WavefrontAlertReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 		if exist && &wfAlert.Status.ExportParamsChecksum != nil {
 			if reqChecksum != wfAlert.Status.ExportParamsChecksum {
 				proceed = false
+				status.State = alertmanagerv1alpha1.ReadyToBeUsed
+				status.ExportParamsChecksum = reqChecksum
 			}
 		}
 	}
+	// Validate the alert request
+	var alert wf.Alert
+	r.convertAlertCR(ctx, &wfAlert, &alert)
+	if err := wavefront.ValidateAlertInput(ctx, &alert); err != nil {
+		r.Recorder.Event(&wfAlert, v1.EventTypeWarning, err.Error(), "alert input request validation failed")
+		state := alertmanagerv1alpha1.Error
+		log.Error(err, "alert input request validation failed")
+		wfAlert.Status = alertmanagerv1alpha1.WavefrontAlertStatus{
```

Comment:
If the validation was failed, we are dropping all the individual alerts here. This makes sense if it is the first time create alert. If the existing alert type CR already have multiple alerts associated with it, and validation was failed for a user change. we are going to lose the associated alerts?

Reply:
You mean we are overwriting the individual alerts status? Yeah- Thats a good point. Let me update that

Reply:
yeah, here you overwrite the entire status with a new WavefrontAlertStatus and excluded individual alerts, the existing individual alerts will be dropped